File: 2. Add Two Numbers/py/AddTwoNumbers.py
# LeetCode challenge: 2. Add Two Numbers
# <https://leetcode.com/problems/add-two-numbers/description/>


class Solution:
    def addTwoNumbers(self, l1: list[int], l2: list[int]) -> list[int]:
        return self.int_to_list(self.list_to_int(l1) + self.list_to_int(l2))

    def list_to_int(self, l1: list[int]) -> int:
        number = 0
        for n in range(len(l1)):
            number += l1[n] * (10 ** n)
        return number

    def int_to_list(self, num: int) -> list[int]:
        s_num = str(num)
        len_s = len(s_num)
        li = []
        for i in range(len_s):
            li.append(int(s_num[len_s - i - 1]))
        return li

File: 2. Add Two Numbers/cs/AddTwoNumbers/AddTwoNumbers.cs
/*
LeetCode Problem 2. Add Two Numbers
https://leetcode.com/problems/add-two-numbers/description/
*/

// Definition for singly-linked list.
// public class ListNode {
//     public int val;
//     public ListNode next;
//     public ListNode(int val=0, ListNode next=null) {
//         this.val = val;
//         this.next = next;
//     }
// }

namespace LeetCode.AddTwoNumbers;

public class Solution
{
    public static int[] AddTwoNumbers(int[]? list1, int[]? list2)
    {
        return IntToList(ListToInt(list1) + ListToInt(list2));
    }

    private static long ListToInt(int[]? list)
    {
        if (list is null) return 0;
        long number = 0;
        for (int n = 0; n < list.Length; n++)
        {
            number += list[n] * (long)Math.Pow(10, n);
        }
        return number;
    }

    private static int[] IntToList(long num)
    {
        string sNum = num.ToString();
        int lenS = sNum.Length;
        int[] result = new int[lenS];
        for (int i = 0; i < lenS; i++)
        {
            result[i] = int.Parse(sNum[lenS - i - 1].ToString());
        }
        return result;
    }
}